### PR TITLE
Fix NPE in machine actuator

### LINKED ISSUE #287: NPE in machine actuator
/kind bug

**What steps did you take and what happened:**
The controller manager outputs the following NPE:
```
I1018 20:54:00.288034       1 controller.go:89] Running reconcile Cluster for test1
I1018 20:54:00.288486       1 controller.go:124] reconciling cluster object test1 triggers idempotent reconcile.
I1018 20:54:00.288494       1 actuator.go:64] Reconciling cluster test1.
I1018 20:54:00.288164       1 controller.go:113] Running reconcile Machine for aws-controlplane-g9nvr
I1018 20:54:00.288858       1 actuator.go:301] Checking if machine aws-controlplane-g9nvr for cluster test1 exists
E1018 20:54:00.289144       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:388
/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine/actuator.go:90
/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine/actuator.go:303
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:157
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:207
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xd1a3e7]

goroutine 110 [running]:
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x1330320, 0x2005a20)
	/usr/local/go/src/runtime/panic.go:502 +0x229
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine.(*Actuator).machineProviderStatus(0xc420588750, 0xc4203f6840, 0xc4201b59b0, 0x2, 0x2)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine/actuator.go:90 +0x47
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine.(*Actuator).Exists(0xc420588750, 0xc4200c4d00, 0xc4203f6840, 0x0, 0x0, 0xc420509a01)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine/actuator.go:303 +0x112
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc4205aa580, 0xc4204b5770, 0x7, 0xc420509aa0, 0x16, 0x3, 0x3, 0x3, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:157 +0x397
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc4204ccb40, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:207 +0x139
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x36
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4200219e0)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4200219e0, 0x3b9aca00, 0x0, 0x1, 0xc42059e360)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4200219e0, 0x3b9aca00, 0xc42059e360)
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:156 +0x388
```

**What did you expect to happen:**
No NPE

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vince@vincepri.com>

**What this PR does / why we need it**:
This PR fixes a NPE in the machine actuator caused by the machine state to be empty on bootstrap. It also removes the state update, which is tracked in the follow-up issue #288.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #287 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ import (
 	"flag"
 	"os"
 
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/apis"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/cluster"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine"
```
Comment (detiber) [inline cmd/manager/main.go]: Not a blocker by any means, but I think there is value in keeping the local imports in a separate group.

Comment (phyber) [reply cmd/manager/main.go]: Looking at the ordering applied there, that might be `gofmt`, which means that local imports should always be grouped together at least, even if not separate.

File: Makefile [hunk 2]
```diff
@@ -28,15 +28,15 @@ endif
...
+	${GOPATH}/bin/dep version || go get -u github.com/golang/dep/cmd/dep
 	${GOPATH}/bin/dep ensure
 
 check-install:
 	./scripts/check-install.sh
 
 depend-update:
 	dep version || go get -u github.com/golang/dep/cmd/dep
-	${GOPATH}/bin/dep ensure -update
+	dep ensure -update
```
[the remark this answers is not in the document]
Comment (vincepri) [reply Makefile]: Good catch, it should be the same, although I think those commands are redundant at this point, will fix.

**Pull request merged**